DOCS-9826: TGW topics (#448)
mulesoft/docs-runtime-manager: modules/ROOT/pages/tgw-add-arm.adoc

= Add Transit Gateway Attachments
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

image:logo-cloud-active.png[link="/runtime-manager/deployment-strategies#cloudhub", title="CloudHub"]
image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies#hybrid", title="Hybrid"]
image:logo-server-disabled.png[link="/runtime-manager/deployment-strategies#anypoint-platform-private-cloud-edition", title="Anypoint Platform PCE"]
image:logo-rtf-disabled.png[link="/runtime-manager/deployment-strategies#anypoint-runtime-fabric", title="Runtime Fabric"]

> Use Runtime Manager to add a transit gateway and connect your MuleSoft Virtual Private Cloud (VPC) to your private corporate network.

Reviewer: I would use the full name first time in each Level 1 topic, but it's up to you:
```suggestion
Use Anypoint Runtime Manager to add a transit gateway and connect your MuleSoft Virtual Private Cloud (VPC) to your private corporate network.
```

== Prerequisites

Before you can add a transit gateway:

* Purchase an Anypoint VPC offering with Transit Gateway entitlements.
+
The *Transit Gateways* page in Runtime Manager is visible to Anypoint organizations that have Transit Gateway entitlements. 
* Sign in to an Anypoint Platform account with either the CloudHub Network Administrator or the CloudHub Network Viewer user permission.
+
See xref:access-management::users.adoc#granting-permissions-and-roles-to-users[Granting Permissions and Roles to Users].

> * Set your default environment (in your profile) to an environment with Read Application permissions.

Reviewer: Alternative:
```suggestion
* In your profile, set your default environment to an environment with Read Application permissions.
```

Reviewer 2: Is it really necessary to set the default environment? TGWs live at the root BG, so they don't have direct relationships with environments.

Author: Removed.

+
See xref:access-management::managing-your-account.adoc#update-user-information[Update User Information].
* Create an Anypoint VPC in Anypoint Platform.
+
See xref:vpc-tutorial.adoc[Create and Update a VPC for Runtime Manager].
* Create a transit gateway on your corporate AWS account.
+

> You must have access to this AWS account to add a transit gateway.

Reviewer 2: This threw me off at first, and then once I understood it, it seemed redundant. Might just be me though.

Author: Removed redundancy.

+
See https://docs.aws.amazon.com/vpc/latest/tgw/tgw-getting-started.html[Getting started with transit gateways^] in the AWS documentation for information about creating a transit gateway on AWS.
+

> One transit gateway can support up to 10 VPC connections.

Reviewer 2: This actually would be an appropriate place to say "VPC attachments".

Author: changed.

* Identify the subnets in your network (in CIDR notation) that you want to make accessible through the transit gateway.

> == Overview 

Reviewer: I wonder if it would be more precise to title this something with the work "Workflow" in it, rather than "Overview," unless you want to specify that this is the "Task Overview"?

Author: Yes.

> The steps for adding a Transit Gateway to your VPC are:

Reviewer: Transit Gateway is lowercase in similar contexts earlier (lines 12, 31, 35, 37, 38).
If this is a proper noun, it should be followed by a common noun because it is preceded by "a".

Author: fixed

> . <<add_tgw,Add the transit gateway>> in Anypoint Platform.

Reviewer 2: IMO it's odd to represent these as two items, because they add the TGW to Anypoint _by_ creating a resource share (and verifying it).

> . <<create_resource_share,Create a Resource Share in AWS>> to share your transit gateway with Anypoint Platform.

Reviewer: Per Amazon:
```suggestion
. <<create_resource_share,Create a resource share in AWS>> to share your transit gateway with Anypoint Platform.
```

. <<attach_vpc_tgw,Attach your VPCs>> to the transit gateway.

> . <<accept_attachment,Accept the VPC attachments>> in AWS.

Reviewer 2: They attach their VPCs to the TGW by…

1. Identifying a VPC to attach (MuleSoft then automatically creates the attachment)
2. Accepting the attachment MuleSoft creates
3. Repeating as necessary for other VPCs

I think you can still say "Attach your VPCs to the transit gateway," but in that case I wouldn't list "Accept the VPC attachments" as a separate step.

Author: @taylordolan - I separated them out because they're two different processes (Anypoint side and AWS side)/sections.

I changed it to combine them into a single step in this list.

. <<enable_outbound_traffic_from_vpc,Enable outbound traffic from the Anypoint VPC>>. 
. <<enable_inbound_traffic_on_aws,Enable inbound traffic through the transit gateway on AWS>>.

[[add_tgw]]
== Add a Transit Gateway

To add a transit gateway:

. Sign in to Anypoint Platform and select *Runtime Manager*.

> . Switch to the environment where you want to add a transit gateway.

Reviewer 2: This isn't true. They might have to select an environment because our UI enforces it (unfortunately), but the TGW is added to the root business group. (Or maybe it's added to the selected BG? I can't remember.)

+

> See xref:runtime-manager-switch-env.adoc[Use Different Environments in Runtime Manager].

Reviewer: Just FYI, if you _want_ to have a two-sentence bullet list item, I have eased the bullet list style recommendation....

. From the Runtime Manager navigation menu, click *Transit Gateways*, and then click *Add Transit Gateway*.
+
The *Here's what's coming* page lists the steps to add the transit gateway:
+
_update screenshot - styles changed_
+
image::tgw-process-overview.png[Here's what's coming page]
. Click *I'm ready*. 
. Specify a name for your transit gateway:
+
_update screenshot - styles changed_
+
image::tgw-add.png[Add transit gateway page]
+
Use the same name for your transit gateway in AWS.
You can change this name later.
The name can contain up to 255 alphanumeric characters (a-z, A-Z, 0-9) and hyphens (-).
// They cannot start or end with a hyphen and cannot contain spaces or other characters.
. Select the region that corresponds to the location of your AWS transit gateway.
+
See xref:cloudhub-networking-guide.adoc#regional-services[Regional Services].
+
Your VPC and AWS transit gateway must be in the same region.
. Click *Next*.
+
The *Add transit gateway* page lists the steps to create a resource share in AWS:
+
_update screenshot - styles changed_
+
image::tgw-add-overview.png[Add transit gateway page: Steps to create a resource share]
+
[NOTE]

> The MuleSoft AWS account ID that appears depends on your platform (US, EU, or Government Cloud).

Reviewer: We must always use "MuleSoft" before "Government Cloud," to avoid confusion with the Federal, Microsoft version:
```suggestion
The MuleSoft AWS account ID that appears depends on your platform (US, EU, or MuleSoft Government Cloud).
```

[[create_resource_share]]
== Create a Resource Share

. Sign in to your AWS corporate account.
. In Anypoint Platform, click the *Create resource share* link on the *Add transit gateway* page. 
+
The link opens the AWS RAM console to the page for creating a resource share in the region you specified.

> . On AWS, on the *Create resource share* page:

Reviewer: Will this work as well?
```suggestion
. On the AWS *Create resource share* page:
```

+
--
.. Under *Description*, enter a descriptive name for the resource share in the *Name* field.
.. Under *Resources - _optional_*, select *Transit Gateways* from the *Select resource type* menu and select the transit gateway resource to share.
+
The transit gateway ID appears in the *Selected resources* field.
.. Under *Principals - _optional_*, ensure that *Allow external accounts* is selected, 
enter the MuleSoft AWS account ID that appears on the *Add transit gateway* page, and click *Add*.
+
The AWS account number appears in the *Selected principals* field.
.. Under *Tags*, add a tag if you want.
.. Click *Create resource share*.

> .. Copy the following values for the resource share you just created: *ID* and *Owner*.

Reviewer: ...and this?
```suggestion
.. Copy the *ID* and *Owner* values for the resource share you just created. 
```

--
. In Anypoint Platform, in the *Add transit gateway* window: 
+
--
.. Click *Next*.
.. Paste the values you copied from AWS in the *ID* and *Owner* fields:
+
_update screenshot - styles changed_
+
image::tgw-add-resource-share.png[Add transit gateway page: Resource share ID and owner ID]

*** The resource share *ID* field contains alphanumeric characters (a-z, A-Z, 0-9) and hyphens (-).
*** The resource share *Owner* field contains only numbers.

.. Click *Add*.
--
+
The *Transit Gateways* page shows the progress for connecting to AWS and adding the transit gateway to Anypoint Platform.

When the transit gateway addition succeeds, the *Transit Gateways* page displays the *Transit gateway added* message.
The *Transit Gateways* page updates the owner and ID from AWS and the state as *Available*:

_Update screenshot - title and contents changed_

image::tgw-resource-share-available.png[Transit gateway added]

If the transit gateway addition fails, see xref:tgw-troubleshoot.adoc[Troubleshoot Transit Gateway Attachments].

[[attach_vpc_tgw]]
== Attach a VPC to the Transit Gateway

> After the transit gateway attachment succeeds, you can attach your Anypoint VPC to enable apps to access the transit gateway.

Reviewer: To omit the modal "can":
```suggestion
After the transit gateway attachment succeeds, attach your Anypoint VPC to enable apps to access the transit gateway.
```

To attach a VPC to the transit gateway:

. In Anypoint Platform, on the *Transit Gateways* page, click *Attach VPC*.
+
The *Attach VPC* button is disabled if no VPCs are available to attach or if the transit gateway addition failed. 
. From the *Select VPC* window, select the VPC to attach.
+

> _Add screenshot?_

Reviewer: If so, end previous with a colon.

+
The window displays only the VPCs that are:
+
--
* In the same region as the transit gateway

> * Not already attached to a transit gateway

Reviewer 2: I had assumed VPCs can be attached to multiple TGWs. I don't actually know if that's true, but if it is, we should say "the transit gateway" here.

Author: @taylordolan - In your design spec, you say VPCs "That have not already been added."

Reviewer 2: A VPC can't be attached the the same TGW multiple times, but it can be attached to multiple TGWs.

> * In Business Groups for which you have Cloudhub Network Administrator permissions

Reviewer: Does "Business Groups" require the init. caps.?

Author: Nope. Fixed.

--
. Click *Next*.

[[accept_attachment]]
== Accept the VPC Attachment

. Sign in to your AWS corporate account.
. In Anypoint Platform, on the *Accept VPC Attachment* page, click the *Transit Gateway Attachments* link.
+

> _Add screenshot?_

Reviewer: ditto colon

+
The link opens the AWS RAM console to the *Create Transit Gateway Attachment* page.
. In the AWS console:
+
.. Select the attachment that shows *pending acceptance* in the *State* column.
+
The attachment might take a few minutes to appear.
+

> .. Select the Transit Gateway attachment ID and, on the *Details* tab, ensure that the *Resource owner account ID* is the MuleSoft AWS account ID.

Reviewer: ```suggestion
.. Select the Transit Gateway attachment ID and, on the *Details* tab, ensure that *Resource owner account ID* is the MuleSoft AWS account ID.
```

> .. Click the pencil icon for the attachment and enter a name. 

Reviewer: Can you add an icon capture rather than describing it?

Reviewer 2: I'm not sure what this refers to. Is this part of the AWS UI?

Author: @taylordolan - yes.

.. Select *Actions* > *Accept* and click *Accept* to confirm.
+
When the attachment acceptance succeeds, the *State* column shows *available*.
. In Anypoint Platform, in the *Accept VPC Attachment* window, click *Done*.
+
The *Transit Gateways* page shows the progress for accepting the VPC attachment from Anypoint Platform to AWS.
+

> _Add screenshot?_

Reviewer: ditto colon

. Click *Refresh* to update the attachment status.

When the VPC attachment succeeds, the *Transit Gateways* page displays the *VPC attached* message and the attachment state indicates that it's attached to Transit Gateway:

_Update screenshot - details and styling changed_

image::tgw-attached.png[VPC attached successfully]

> 

Reviewer: Remove extra line.

If the VPC attachment state is `Rejected`, see xref:tgw-troubleshoot.adoc[Troubleshoot Transit Gateway Attachments].

[[enable_outbound_traffic_from_vpc]]
== Enable Outbound Traffic from the VPC

After attaching the VPC, enable outbound traffic from the VPC to an external destination so that apps can access the transit gateway and the transit gateway routes traffic correctly.

> To do this, add routes to the transit gateway route table.
> 
> To add routes to the transit gateway route table:

Reviewer: ```suggestion
To do this, add routes to the transit gateway route table:
```

. In Anypoint Platform, on the *Transit Gateways* page, in the *VPCs* section, click *Add Route*. 

> . On the *Add VPC Route* window, enter a single subnet in CIDR notation for your

Reviewer: ```suggestion
. On the *Add VPC Route* window, use CIDR notation to enter a single subnet for your
```

AWS VPC:
+
_Update screenshot - title, text, and style changed_
+
image::tgw-add-vpc-route.png[Add VPC route]
+

> For information, see xref:tgw-about.adoc#routing[Transit Gateway Routing].

Reviewer: For what kind of information? "further"? "about..."?
